src/pure/List.mo

```diff
@@ -686,7 +686,7 @@ module {
   /// Space: O(1)
   ///
   /// *Runtime and space assumes that `equalItem` runs in O(1) time and space.
-  public func equal<T>(list1 : List<T>, list2 : List<T>, equalItem : (T, T) -> Bool) : Bool = switch (list1, list2) {
+  public func equal<T>(list1 : List<T>, list2 : List<T>, equalItem : (implicit : (T, T) -> Bool)) : Bool = switch (list1, list2) {
```

Comment:
```suggestion
  public func equal<T>(list1 : List<T>, list2 : List<T>, equalItem : (implicit (equal : (T, T) -> Bool))) : Bool = switch (list1, list2) {
```
This overrides the local name equalItem to resolve to `equal` (otherwise it will look for equalItem and find nothing.